src/Types/Node.hs

```diff
@@ -49,39 +46,144 @@ data Node = Node
   -- ^ mainnet01, testnet04, etc or Nothing if it's a pact -s server
   }
 
-getNode :: LogEnv -> HostPort -> IO (Either String Node)
-getNode le h = do
-    httpsMgr <- newTlsManagerWith (mkManagerSettings (TLSSettingsSimple True False False) Nothing)
-    let httpsUrl = "https://" <> infoUrl
-    req <- parseRequest httpsUrl
-    logFLE le DebugS req "getNode: trying chainweb https"
-    try @_ @SomeException (httpLbs req httpsMgr) >>= \case
-        Right resp | statusIsSuccessful (responseStatus resp) -> do
-            case eitherDecode (responseBody resp) of
-              Left e -> return $ Left ("Error decoding HTTPS response: " <> e)
-              Right ni -> do
-                return $ Right $ Node Https h httpsMgr ChainwebServer (Just ni)
-        _ -> do
-            httpMgr <- newManager defaultManagerSettings
-            let httpUrl = "http://" <> infoUrl
-            req2 <- parseRequest httpUrl
-            logFLE le DebugS req2 "getNode: trying chainweb http"
-            resp2 <- httpLbs req2 httpMgr
-            if statusIsSuccessful (responseStatus resp2)
-              then case eitherDecode (responseBody resp2) of
-                Left e -> return $ Left ("Error decoding HTTP response: " <> e)
-                Right ni -> return $ Right $ Node Http h httpMgr ChainwebServer (Just ni)
-              else do
-                req3 <- parseRequest ("https://" <> infoUrl)
-                logFLE le DebugS req3 "getNode: trying pact server"
-                resp3 <- httpLbs req3 httpMgr
-                if statusIsSuccessful (responseStatus resp3)
-                  then return $ Right $ Node Https h httpMgr PactServer Nothing
-                  else return $ Left ("Error requesting from " <> infoUrl)
+getNodeServiceApi :: LogEnv -> SchemeHostPort -> IO (Either String Node)
+getNodeServiceApi le shp = do
+    -- TODO Add exception handling from below comment & delete comment
+
+    let hp = _shp_hostPort shp
+    case _shp_scheme shp of
+      Nothing -> runExceptT $
```

Comment:
Ah, it means auto-select. Maybe better to have that as an explicit choice of Scheme?

Reply:
The main thing this PR does is introduce the ability for the user to choose explicitly while also keeping support for not specifying to preserve backwards compatibility and smooth the user experience.